Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -323,27 +324,22 @@ calculateFinalValuesForCorrectDistanceCalculations handle booking ride now mbMax
   if not pickupDropOutsideOfThreshold
     then
       if distanceDiff > thresholdConfig.actualRideDistanceDiffThreshold
-        then recalculateFareForDistance handle booking ride (roundToIntegral $ min ride.traveledDistance maxDistance)
+        then recalculateFareForDistance handle booking ride (roundToIntegral $ min ride.traveledDistance maxDistance) False
         else returnEstimatesAsFinalValues handle booking ride
     else
       if distanceDiff < 0
-        then do
-          fare1@(_, money1, _) <- recalculateFareForDistance handle booking ride (roundToIntegral $ ride.traveledDistance + abs (distanceDiff * 0.5))
-          (dist2, money2', fp2) <- recalculateFareForDistance handle booking ride $ roundToIntegral ride.traveledDistance
-          let money2 = money2' + 50 - (fromMaybe 0 booking.fareParams.driverSelectedFare)
-          let fare2 = (dist2, money2, fp2)
-          return $
-            if money1 < money2
-              then fare1
-              else fare2
+        then
+          if ride.traveledDistance > metersToHighPrecMeters thresholdConfig.pickupLocThreshold
```

Comment:
why are we comparing `traveledDistance` with pickupLocThreshold?

Reply:
Because we want to apply the penalty only if the rider has travelled at least 500m.